app/controllers/api/v0/admins_controller.rb

```diff
@@ -18,8 +18,14 @@ def add_group
       end
 
       def add_workflow
+        group_ids = params.require(:group_ids)
+        groups = Group.find(group_ids)
+
         workflow = Template.find(params.require(:template_id)).workflows.create!(workflow_params)
+        workflow.groups << groups
         json_response(workflow, :created)
+      rescue ActiveRecord::RecordNotFound => e
+        json_response({ :message => e.message }, :unprocessable_entity)
```

Comment:
with your enhancement in `exception_handler` do you still need this rescue?

Reply:
Yes, `exception_handler` is the overall handler. Here when the input group ids are invalid, we want to return error code 422, rather than 404 of RecordNotFound.